packages/yarnpkg-pnpify/sources/generateSdk.ts

```diff
@@ -1,10 +1,129 @@
+import {Report, MessageName}                       from '@yarnpkg/core';
 import {Filename, PortablePath, npath, ppath, xfs} from '@yarnpkg/fslib';
+import {parseSyml, stringifySyml}                  from '@yarnpkg/parsers';
 import {PnpApi}                                    from '@yarnpkg/pnp';
 import chalk                                       from 'chalk';
-import CJSON                                       from 'comment-json';
+import {UsageError}                                from 'clipanion';
+import {startCase, capitalize}                     from 'lodash';
 
 import {dynamicRequire}                            from './dynamicRequire';
 
+import {BASE_SDKS}                                 from './sdks/base';
+import {VSCODE_SDKS}                               from './sdks/vscode';
+
+export const PNPIFY_FOLDER = `.yarn/pnpify` as PortablePath;
+
+export const SUPPORTED_EDITORS = new Set([
+  `vscode` as const,
+]);
+
+export const getDisplayName = (name: string) =>
+  startCase(name).split(` `).map(word => capitalize(word)).join(` `);
+
+export const validateEditors = (editors: Set<string>) => {
+  const unsupportedEditors: Array<string> = [];
+
+  for (const editor of editors) {
+    if (!SUPPORTED_EDITORS.has(editor as SupportedEditor)) {
+      unsupportedEditors.push(editor);
+    }
+  }
+
+  if (unsupportedEditors.length > 0) {
+    throw new UsageError(`No supported editors with the following names could be found: ${unsupportedEditors.join(`, `)}`);
+  }
+};
+
+export type SetTemplate<S> = S extends Set<infer T> ? T : never;
+
+export type SupportedEditor = SetTemplate<typeof SUPPORTED_EDITORS>;
+
+export class EditorsFile {
+  public __metadata: {
+    // The version of the format.
+    // Not currently used for anything,
+    // might be useful for third-parties to detect
+    // if different fields are supposed to exist inside the file
+    version: number | null;
+    // Might be used in the future to decide whether we
+    // actually need to update the SDKs or not
+    pnpifyVersion: string | null;
+  } = {
+    version: null,
+    pnpifyVersion: null,
+  };
```

Comment:
I think I'd prefer to keep it simple for now and remove the versions. I see a lot of code around this logic that *may* be useful in the future, but there's no reason why we can't just say later on that the lack of metadata is a version indicator in itself if we need to.